### ISSUE TITLE
The position of the tracking box and the target does not match

### ISSUE THREAD

**Issue by one-cai-ji** [Smorodov/Multitarget-tracker#282]
![1](https://user-images.githubusercontent.com/63446384/104808114-02d9f100-581f-11eb-939e-c7685c6cbc2b.png)

Hi, I used my training weight to replace the weight in the carscounting example, but the position deviation between the tracking box and the target object is large, and It performed well when using yolov4 (Darknet) detection. What can I do to improve the corresponding relationship between the tracking box and the target object

**Comment by Nuzhny007** [comment]
Hi!
Do you can show a small video with detection and tracking that I will understand the problem? The first idea - changing Kalman filter settings but I don't sure.

**Comment by one-cai-ji** [comment]
Hi, I updated the video demo of detection and tracking


https://user-images.githubusercontent.com/63446384/104865835-07fe8380-5978-11eb-856d-df806ebc246c.mp4



https://user-images.githubusercontent.com/63446384/104866208-f23d8e00-5978-11eb-90a3-4e237c78b4d0.mp4

**Comment by Nuzhny007** [comment]
hm, it need some experiments....

1. Try to increase dt parameter in Kalman Filter:
settings.m_dt = 0.4f;                           // Delta time for Kalman filter 

2. Change IsRobust criteria:
if (track.IsRobust(cvRound(2),          // Minimal trajectory size
                                0.5f,                        // Minimal ratio raw_trajectory_points / trajectory_lenght
                                cv::Size2f(0.1f, 8.0f))

**Comment by one-cai-ji** [comment]
Hi, can I use real-time video captured by camera instead of local video

**Comment by Nuzhny007** [comment]
Yes. Just set ip-address or web camera index instead path to video file

**Comment by one-cai-ji** [comment]
Your reply has helped me a lot.
I want to change the weight and configuration file in CarCounting to yolov4tiny version. However, there is an error in CreateDetector. What can I do to solve this problem

**Comment by one-cai-ji** [comment]
sorry,I found It is my problem. The problem caused by one more space in the modified path ,it is not the source code problem

**Comment by one-cai-ji** [comment]
Hello, I try to use the algorithm to test the video. The algorithm performs well for objects with low density, but for objects with high density, the matching error rate is very high. What can I do to improve this phenomenon

**Comment by Nuzhny007** [comment]
So, detector works good but the tracker can't work accuracy? The objects ids "jumping" from one object to another when objects are occluded?

**Comment by one-cai-ji** [comment]
![t](https://user-images.githubusercontent.com/63446384/111102536-49ca3580-8587-11eb-8bcb-5b833e8569a4.png)
![d](https://user-images.githubusercontent.com/63446384/111102543-4f278000-8587-11eb-96e4-e79f10220427.png)
I uploaded two images, as shown in the picture. I think the detector works well, but some targets with more overlapped are not tracked

**Comment by Nuzhny007** [comment]
A classic way to resolve this problem - using re-identification. For example DeepSORT or any similar. So are some pretrained models for faces or pedestrians or vehicles. In this project this case can work with model from OpenVINO:  https://github.com/Smorodov/Multitarget-tracker/blob/master/example/examples.h#L683
We add model for calculate embeddings and set object type: m_trackerSettings.m_embeddings.emplace_back(...)
I'm tested this code for pedestrians and vehicles. But I don't sure how this approach will work for fishes. And you need train re-id model for fishes...

May be we can use visual objects tracker on each frame for more robust tracking. So I'll write test for this case in the coming days.
